src/NUnitEngine/nunit.engine.api/NUnitEngineException.cs

```diff
@@ -38,23 +39,45 @@ namespace NUnit.Engine
 #endif
     public class NUnitEngineException : Exception
     {
+        private const string AggregatedExceptionsMsg =
+            "Multiple exceptions encountered. Retrieve AggregatedExceptions property for more information";
+
         /// <summary>
         /// Construct with a message
         /// </summary>
-        public NUnitEngineException(string message) : base(message) { }
+        public NUnitEngineException(string message) : base(message)
+        {
+        }
 
         /// <summary>
         /// Construct with a message and inner exception
         /// </summary>
         /// <param name="message"></param>
         /// <param name="innerException"></param>
-        public NUnitEngineException(string message, Exception innerException) : base(message, innerException) { }
+        public NUnitEngineException(string message, Exception innerException) : base(message, innerException)
+        {
+        }
+
+        /// <summary>
+        /// Construct with a message and collection of inner exceptions
+        /// </summary>
+        public NUnitEngineException(ICollection<Exception> aggregatedExceptions) 
+            : base(AggregatedExceptionsMsg, new NUnitEngineException(AggregatedExceptionsMsg))
+        {
+            AggregatedExceptions = aggregatedExceptions;
+        }
```

Comment:
Not clear what you are doing here. Seems like you end up with an `NUnitEngineException` with an `InnerException` that's also an `NUnitEngineException`, both having the same message. The outer also gets a collection of aggregated exceptions. What's the point of the `InnerException`?

Reply:
I think I was maybe thinking of `NUnitException` in the framework, where I thought we'd always expect an inner exception? I'll get rid of it.